packages/backend/src/open_payments/client/service.ts

```diff
@@ -1,50 +1,146 @@
+      connection.assetCode,
+      connection.assetScale
+    )
+  }
+
+  public get asset(): AssetOptions {
+    return {
+      code: this.assetCode,
+      scale: this.assetScale
+    }
+  }
+
+  public get incomingAmount(): Amount | undefined {
+    if (this.incomingAmountValue) {
+      return {
+        value: this.incomingAmountValue,
+        assetCode: this.assetCode,
+        assetScale: this.assetScale
+      }
+    }
+    return undefined
+  }
+
+  public get receivedAmount(): Amount | undefined {
+    if (this.receivedAmountValue !== undefined) {
+      return {
+        value: this.receivedAmountValue,
+        assetCode: this.assetCode,
+        assetScale: this.assetScale
+      }
+    }
+    return undefined
+  }
+
+  public toResolvedPayment(): ResolvedPayment {
+    return {
+      destinationAsset: this.asset,
+      destinationAddress: this.ilpAddress,
+      sharedSecret: this.sharedSecret,
+      requestCounter: Counter.from(0)
+    }
+  }
+}
+
 export interface OpenPaymentsClientService {
-  incomingPayment: {
-    get(url: string): Promise<IncomingPaymentJSON | undefined>
+  receiver: {
+    get(url: string): Promise<Receiver | undefined>
   }
 }
 
-export interface ServiceDependencies extends BaseService {
+interface ServiceDependencies extends BaseService {
   accessToken: string
   openApi: OpenAPI
-  validateResponse: ValidateFunction<IncomingPaymentJSON>
+  validateConnection: ValidateFunction<ConnectionJSON>
+  validateIncomingPayment: ValidateFunction<IncomingPaymentJSON>
 }
 
 export async function createOpenPaymentsClientService(
-  deps_: Omit<ServiceDependencies, 'validateResponse'>
+  deps_: Omit<
+    ServiceDependencies,
+    'validateConnection' | 'validateIncomingPayment'
+  >
 ): Promise<OpenPaymentsClientService> {
   const log = deps_.logger.child({
     service: 'OpenPaymentsClientService'
   })
-  const validateResponse =
-    deps_.openApi.createResponseValidator<IncomingPaymentJSON>({
-      path: '/incoming-payments/{incomingPaymentId}',
-      method: HttpMethod.GET
-    })
   const deps: ServiceDependencies = {
     ...deps_,
     logger: log,
-    validateResponse
+    validateConnection: deps_.openApi.createResponseValidator<ConnectionJSON>({
+      path: '/connections/{connectionId}',
+      method: HttpMethod.GET
+    }),
+    validateIncomingPayment:
+      deps_.openApi.createResponseValidator<IncomingPaymentJSON>({
+        path: '/incoming-payments/{incomingPaymentId}',
+        method: HttpMethod.GET
+      })
   }
   return {
-    incomingPayment: {
-      get: (url) => getIncomingPayment(deps, url)
+    receiver: {
+      get: (url) => getReceiver(deps, url)
     }
   }
 }
 
-export async function getIncomingPayment(
+async function getConnection(
   deps: ServiceDependencies,
   url: string
-): Promise<IncomingPaymentJSON | undefined> {
+): Promise<ConnectionJSON | undefined> {
   const requestHeaders = {
     Authorization: `GNAP ${deps.accessToken}`,
```

Comment:
Is this necessary?

Reply:
There is no auth for getting a connection.